x-pack/legacy/plugins/spaces/mappings.json

```diff
@@ -22,6 +22,12 @@
       "disabledFeatures": {
         "type": "keyword"
       },
+      "imageUrl": {
+        "type": "text"
```

Comment:
I talked to the Elasticsearch engineering team about what we are trying to do here, and I have a rather invasive proposal for you:

[`text` fields](https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html#text) end up being analyzed when they are indexed, which adds processing time and other overhead whenever a space is created, updated, or searched for. Since the image url is mostly a base-64 blob of data, we shouldn't need to incur this overhead, as we will never be searching or aggregating on this field.

The most effecient way to store base-64 encoded data in Elasticsearch is to use the [`binary` field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/binary.html). The binary field type requires that data be base-64 encoded when you store it, which is _almost_ perfect for this use case. The vast majority of the data is already in that format, except for the data-url prefix.

We could just base-64 encocde the entire data url before storing it, but then we would be double-encoding a vast majority of the field, which adds unnecessary processing, and increases the size of the stored field significantly.

Rather than base-64 encoding this entire string again, we're proposing the following:
- Split the imageUrl into two fields: one for `imageType`, and another for `imageData`.
- `imageType` should be a `keyword`, and will store the mime type of the uploaded image.
- `imageData` should be of type `binary`, and will store the base-64 encoded image blob.

Then on the client side, we can construct a data-url on the fly by combining these two fields together witin the `SpaceAvatar` component.

Reply:
@legrego : what about storing the image as it is and making the field not analyzed/not searchable? Would it solve the problem? Remember also that we limited the image size to 64x64, which is few kb encoded.
Thanks.